test/Test_Menhir.re

```diff
@@ -0,0 +1,458 @@
+    Constructor("A", Int |> Typ.fresh) |> Exp.fresh,
+    "A ~ Int",
+  ),
+  // TODO Fix for the tests below
+  skip_menhir_only_test(
+    "Constructor with Type Variable",
+    Constructor("A", Var("T") |> Typ.fresh) |> Exp.fresh,
+    "A ~ T",
+  ),
+  skip_parser_test(
+    "Type Variable",
+    Let(
+      Cast(
+        Var("x") |> Pat.fresh,
+        Var("T") |> Typ.fresh,
+        Unknown(Internal) |> Typ.fresh,
+      )
+      |> Pat.fresh,
+      EmptyHole |> Exp.fresh,
+      Var("x") |> Exp.fresh,
+    )
+    |> Exp.fresh,
+    "let x : T = ? in x",
+  ),
+  parser_test(
+    "Type Alias",
+    TyAlias(Var("x") |> TPat.fresh, Int |> Typ.fresh, Int(1) |> Exp.fresh)
+    |> Exp.fresh,
+    "type x = Int in 1",
+  ),
+  parser_test(
+    "Test",
+    Test(
+      BinOp(Int(Equals), Int(3) |> Exp.fresh, Int(3) |> Exp.fresh)
+      |> Exp.fresh,
+    )
+    |> Exp.fresh,
+    "test 3 == 3 end",
+  ),
+  parser_test(
+    "Filter",
+    Filter(
+      Filter({act: (Eval, All), pat: Int(3) |> Exp.fresh}),
+      Int(3) |> Exp.fresh,
+    )
+    |> Exp.fresh,
+    "eval 3 in 3" // TODO Use other filter commands
+  ),
+  parser_test(
+    "List Concat",
+    ListConcat(
+      ListLit([Int(1) |> Exp.fresh, Int(2) |> Exp.fresh]) |> Exp.fresh,
+      ListLit([Int(3) |> Exp.fresh, Int(4) |> Exp.fresh]) |> Exp.fresh,
+    )
+    |> Exp.fresh,
+    "[1, 2] @ [3, 4]",
+  ),
+  skip_parser_test(
+    "Integer Ops",
+    BinOp(
+      Int(GreaterThanOrEqual),
+      BinOp(
+        Int(Minus),
+        BinOp(
+          Int(Plus),
+          UnOp(Int(Minus), Int(1) |> Exp.fresh) |> Exp.fresh,
+          Int(2) |> Exp.fresh,
+        )
+        |> Exp.fresh,
+        BinOp(
+          Int(Divide),
+          Int(3) |> Exp.fresh,
+          BinOp(
+            Int(Times),
+            Int(4) |> Exp.fresh,
+            BinOp(Int(Power), Int(5) |> Exp.fresh, Int(6) |> Exp.fresh)
+            |> Exp.fresh,
+          )
+          |> Exp.fresh,
+        )
+        |> Exp.fresh,
+      )
+      |> Exp.fresh,
+      Int(8) |> Exp.fresh,
+    )
+    |> Exp.fresh,
+    "-1 + 2 - 3 / 4 * 5 ** 6 >= 8",
+  ), // TODO Add the remaining operators and fix precedence
+  parser_test("Float", Float(1.) |> Exp.fresh, "1."),
+  skip_parser_test(
+    "Float Ops",
+    BinOp(
+      Float(LessThan),
+      BinOp(
+        Float(Plus),
+        Float(2.) |> Exp.fresh,
+        BinOp(
+          Float(Divide),
+          Float(3.) |> Exp.fresh,
+          BinOp(
+            Float(Times),
+            Float(4.) |> Exp.fresh,
+            BinOp(
+              Float(Power),
+              Float(5.) |> Exp.fresh,
+              Float(6.) |> Exp.fresh,
+            )
+            |> Exp.fresh,
+          )
+          |> Exp.fresh,
+        )
+        |> Exp.fresh,
+      )
+      |> Exp.fresh,
+      Float(8.) |> Exp.fresh,
+    )
+    |> Exp.fresh,
+    "2. +. 3. /. 4. *. 5. **. 6. <. 8." // TODO Add the remaining operators. -. is also currently broken
+  ),
```

Comment:
@green726  I added the float test to make sure we don't forget the equivalent float operations. For some reason -. isn't working right now. I marked all the failing tests as `Alcotest.skip` so we don't have to keep commenting them out. We should make sure they're all passing before we merge though.